cockatrice/src/zoneviewwidget.cpp

```diff
@@ -72,47 +72,44 @@ ZoneViewWidget::ZoneViewWidget(Player *_player, CardZone *_origZone, int numberC
     closeButton->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
     QGraphicsProxyWidget *closeButtonProxy = new QGraphicsProxyWidget;
     closeButtonProxy->setWidget(closeButton);
-    
+
     hbox->addItem(titleLabel);
     hbox->addItem(closeButtonProxy);
     QGraphicsLinearLayout *vbox = new QGraphicsLinearLayout(Qt::Vertical);
-    
+
     vbox->addItem(hbox);
-    
+
     if (numberCards < 0) {
-        sortByNameCheckBox = new QCheckBox;
         QGraphicsProxyWidget *sortByNameProxy = new QGraphicsProxyWidget;
-        sortByNameProxy->setWidget(sortByNameCheckBox);
+        sortByNameProxy->setWidget(&sortByNameCheckBox);
         vbox->addItem(sortByNameProxy);
 
-        sortByTypeCheckBox = new QCheckBox;
         QGraphicsProxyWidget *sortByTypeProxy = new QGraphicsProxyWidget;
-        sortByTypeProxy->setWidget(sortByTypeCheckBox);
+        sortByTypeProxy->setWidget(&sortByTypeCheckBox);
         vbox->addItem(sortByTypeProxy);
-    } else {
-        sortByNameCheckBox = 0;
-        sortByTypeCheckBox = 0;
+
+        QGraphicsProxyWidget *pileViewProxy = new QGraphicsProxyWidget;
+        pileViewProxy->setWidget(&pileViewCheckBox);
+        vbox->addItem(pileViewProxy);
```

Comment:
Just checked the qt src. Each item added with additem() is destroyed when vbox is destroyed.